src/serial/multipart-form-data.ts

```diff
@@ -0,0 +1,56 @@
+import { Serialization } from './';
+import { Entry, EntryList } from './entry-list';
+import normalizeNewlines from './normalize-newlines';
+
+const CRLF = '\r\n';
+
+export async function toMultipartFormData(entryList: EntryList): Promise<Serialization> {
+  const boundary = makeBoundaryString();
+  const body = await toMultipartBody(entryList, boundary);
+  return {
+    body,
+    mediaType: `multipart/form-data; boundary="${boundary}"`
+  };
+}
+
+function makeBoundaryString() {
+  const timestamp = Date.now().toString();
+  const padding = Math.floor(Math.random() * 100000).toString();
+  const boundary = `${timestamp}${padding}`;
+  return `${'-'.repeat(50 - boundary.length)}${boundary}`;

```

Comment:
Curious how you came to the 50 - boundary length calculation.

Reply:
Mostly arbitrary for now. I couldn't find a concrete algorithm for generating a boundary string, so this is what I came up with. The only limitations I could find are that 1) the boundary string can't appear in any of the parts and 2) it can be no longer than 70 characters. It isn't perfect, but I figure this is good enough for a first pass. We can always improve it with any significant impact, as well as add a check for containment in the body.

For reference:
- https://datatracker.ietf.org/doc/html/rfc2046#section-5.1
- https://datatracker.ietf.org/doc/html/rfc7578#section-4.1